### PR TITLE
adds netconf shared module

### PR DESCRIPTION
**Pull request by privateip**
New netconf shared lib added to module_utils for sending netconf
messages to remote devices over netconf connection plugin

### CODE REVIEW

**Review by gundalow — CHANGES_REQUESTED** [review]
I believe we agreed that all new files would start `network_`

You'd previously said:
> agreed, we are standardizing on network_ .... will need to move the rest through a deprecation process

File: lib/ansible/module_utils/netconf.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    obj = new_ele('get-config')
+    children(obj, ('source', source))
+    children(obj, ('filter', filter))
+    return send_request(module, obj)
+
+@contextmanager
+def locked_config(module):
+    try:
+        lock(module)
+        yield
```
Comment (gundalow) [inline lib/ansible/module_utils/netconf.py:79]: This doesn't seem to be py2.4 compliant

```
2017-01-27 04:09:40 Compiling ./lib/ansible/module_utils/netconf.py ...
2017-01-27 04:09:40   File "./lib/ansible/module_utils/netconf.py", line 79
2017-01-27 04:09:40     yield
2017-01-27 04:09:40          ^
2017-01-27 04:09:40 SyntaxError: invalid syntax
```

**Comment by privateip** [comment]
Modules that are common libs for all networking will start with `network_`.  This one is specific to netconf (same as say eos or nxos or ios... etc)

**Comment by mattclay** [comment]
@privateip CI failure due to python 2.4 syntax error:

```
2017-01-27 04:09:40 Compiling ./lib/ansible/module_utils/netconf.py ...
2017-01-27 04:09:40   File "./lib/ansible/module_utils/netconf.py", line 79
2017-01-27 04:09:40     yield
2017-01-27 04:09:40          ^
2017-01-27 04:09:40 SyntaxError: invalid syntax
```

**Pull request merged**